### PR TITLE
Progress on certificate validation for #7

### PR DESCRIPTION
**Pull request by rjzak**
Adding AMD certs from their website, unit tests to validate root certs.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -11,6 +11,8 @@ pkcs10 = { git = "https://github.com/npmccallum/formats", branch = "patch", feat
 der = { git = "https://github.com/npmccallum/formats", branch = "patch", features = ["alloc"] }
 x509 = { git = "https://github.com/npmccallum/formats", branch = "patch" }
 p256 = { version = "^0.10.1" }
+sha2 = { version = "^0.10.1" }
+hex-literal = { version = "0.2.2" }
```
Comment (npmccallum) [inline Cargo.toml:15]: These should be `dev-dependencies`.

**Review by npmccallum — CHANGES_REQUESTED** [review]
Also, are the certificates X.509 or the AMD custom cert format?

File: Cargo.toml (hunk 1 above)
Comment (rjzak) [reply Cargo.toml:15]: They are x509 certificates, and I can put `sha2` and `hex-literal` in `dev-dependencies`.

Comment (rjzak) [reply Cargo.toml]: They look to me like regular x509 certificates but now I'm not 100% sure, double-checking.

Comment (npmccallum) [reply Cargo.toml]: @rjzak What does `openssl` do with them?

Comment (rjzak) [reply Cargo.toml]: OpenSSL throws an error:
```
$ openssl x509 -in ask_ark_rome.cert -text -noout
unable to load certificate
140108917658944:error:0909006C:PEM routines:get_name:no start line:../crypto/pem/pem_lib.c:745:Expecting: TRUSTED CERTIFICATE
```
The `file` command doesn't like them either (all show roughly the same):
`ask_ark_rome.cert: firmware 100 v100 (revision 0) \306\313\317\024[1F\364\ \231\321_\356{\023\023Q\023, 0 bytes or less, at 0x0 0 bytes , at 0x0 0 bytes `

I didn't know AMD had their own version of certificates. I'll instead use the cert output from Enarx.

Comment (rjzak) [reply Cargo.toml]: I was able to get the Milan root certs, but the API isn't giving certs for Rome or Naples. The AMD docs has this as the API server which gives out certs (replace Naples with Milan to see it work): https://kdsintf.amd.com/vcek/v1/Naples/cert_chain

**Pull request closed without merging**